dragonfly_grasshopper/src/DF Model.py

```diff
@@ -0,0 +1,66 @@
+# Dragonfly: A Plugin for Environmental Analysis (GPL)
+# This file is part of Dragonfly.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Dragonfly; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Create a Dragonfly Model, which can be translated to Honeybee model and sent
+for simulation.
+-
+
+    Args:
+        _buildings: A list of Dragonfly Building objects to be added to the Model.
+            Note that at least one Building is necessary to make a simulate-able
+            energy model.
+        context_: Optional Dragonfly ContextShade objects to be added to the Model.
+        _north_: A number between 0 and 360 to set the clockwise north
+            direction in degrees. This can also be a vector to set the North.
+            Default is 0.
+        _name_: A name for the Model. If the name is not provided the name
+            "unnamed" will be used.
+        _run: Set to "True" to run the component and create the Dragonfly Model.
+    
+    Returns:
+        report: Reports, errors, warnings, etc.
+        model: A Dragonfly Model object possessing all of the input geometry
+            objects.
+"""
+
+ghenv.Component.Name = "DF Model"
+ghenv.Component.NickName = 'Model'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Dragonfly"
+ghenv.Component.SubCategory = '0 :: Create'
+ghenv.Component.AdditionalHelpFromDocStrings = "1"
+
+
+try:  # import the core dragonfly dependencies
+    from dragonfly.model import Model
+except ImportError as e:
+    raise ImportError('\nFailed to import dragonfly:\n\t{}'.format(e))
+
+try:
+    from ladybug_rhino.togeometry import to_vector2d
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component) and _run:
+    # set a default name
+    name = _name_ if _name_ is not None else 'unnamed'
```

Comment:
Is it ok to have several models with the same name? I assume it will be fine as we don't have the concept of several models merging into a something different like site yet and we may never do that but just to be safe it might be good to enforce a unique name if the name is not provided by user.

Reply:
If you feel strongly about this, then we should also change the default on the [Honeybee Model as well](https://github.com/ladybug-tools/honeybee-grasshopper-core/blob/master/honeybee_grasshopper_core/src/HB%20Model.py#L64).  However, I an not a big fan of this for the following reasons:
1. There was never really a case in honeybee-legacy where we wanted to generate a unique name for a Model (and all of the resulting simulation folders) where we also didn't want to control what that name was. Thinking of situations like long parametric runs with hundreds of models where you wanted each model to have a unique name / simulation folder, it was often much better to create the unique file name ourselves from the inputs of the run so that we could always figure out later what those original inputs were from the simulation folder.
2. Overwriting the same Model name / simulation folder name was much more preferable over generating a new unique name for each and every new Model.  If we didn't do this, I could see that my simulation folder would quickly become filled with auto-generated garbage names, requiring me to clean it everyone in a while rather than "gleaning as I go" with a standard default name for the Models I didn't care about (eg. "unnamed").

Reply:
I think we are talking about two different concerns.

I agree that a unique name for the same model is the right approach. What I'm trying to avoid is to generate different models with the same name.

The name of the folder should be / will be based on "project name" and then each "run" will have its own name. You can have the same model, which can be part of different projects and use it in different runs. In the legacy plugin there was no such a thing as model so this never came up.

Thinking about it more maybe a better solution is to make model name as a required input. That will shift the responsibility to the user and will also give us a chance to educate them about the importance of having a unique name for a model if possible.

Reply:
I can see more clearly what you are thinking about now and I also see that this issue is all tangled up in the question of how we name/write the folders because I currently use the model name to inform the energy simulation folder name and this has been very useful.  But I haven't thought through simulation folder structure as much as you have and so I think that the best course of action for now is for me to wait and see how you implement this all with Radiance.  Then I can go back here and have the energy simulation follow similar conventions.   For the sake of not breaking SOM's testing, I'd rather keep this as it is now until we make the major change altogether (since anything that I do now is still going to be temporary anyway).

The one thing I would ask, though, is that there be a default name.  When I add up all of the minutes of my own time it will save in my own files and having everyone plug in a name in my workshops, having a default is worth it.

Reply:
This said, I am fine with the default being really unsavory, which urges people to change it (like the default hot pink preview color in Grasshopper)

Reply:
👍 Let's discuss this later.